### PR TITLE
Fix missing RP selection causing 500 error on Landing Page

### PR DESCRIPTION
**Pull request by scruti**
[Jira ticket](https://regulatorydelivery.atlassian.net/browse/COSBETA-1234)
Submit users with multiple Responsible Persons, none of them selected as their current and not having 2FA in their session were getting a 500 code error when reaching the Submit landing page.

From what I have manually tested (and our integration/feature specs), the create or join responsible person check should be triggered even if the users have 2FA enabled and haven't 2FA present in their session.

### CODE REVIEW

File: cosmetics-web/app/controllers/concerns/responsible_person_concern.rb [hunk 1]
```diff
@@ -42,18 +42,6 @@ def validate_responsible_person
...
-    if Rails.configuration.secondary_authentication_enabled
-      current_user&.has_completed_registration? && secondary_authentication_present_in_session?
-    else
-      current_user&.has_completed_registration?
-    end
-  end
-
-  def current_user_fully_registered?
-    current_user && current_user.account_security_completed?
-  end
```
Comment (scruti) [inline cosmetics-web/app/controllers/concerns/responsible_person_concern.rb:55]: Unused code that probably should have been deleted in the past.

File: cosmetics-web/app/controllers/concerns/responsible_person_concern.rb [hunk 2]
```diff
@@ -42,18 +42,6 @@ def validate_responsible_person
 
 private
 
-  def fully_signed_in_submit_user?
-    if Rails.configuration.secondary_authentication_enabled
-      current_user&.has_completed_registration? && secondary_authentication_present_in_session?
-    else
-      current_user&.has_completed_registration?
-    end
-  end
```
Comment (scruti) [inline cosmetics-web/app/controllers/concerns/responsible_person_concern.rb:51]: I have tried to identify a case where this extra check on having the 2FA present in the session was necessary. 
Registered a new user from scratch and from a Responsible Person team invitation to see if we were getting redirected to the "Create or Join RP" page before we had the chance to fill the 2FA code when needed.
Everything seems to work ok. 

If this was needed for some edge case... is not tested. So will trust our feature specs and the basic checks done manually and considered this unnecessary. I hope I'm not wrong!

File: cosmetics-web/app/controllers/submit/landing_page_controller.rb [hunk 3]
```diff
@@ -3,13 +3,20 @@ class Submit::LandingPageController < SubmitApplicationController
   skip_before_action :ensure_secondary_authentication
   skip_before_action :require_secondary_authentication
   before_action :set_responsible_person
+
```
Comment (scruti) [inline cosmetics-web/app/controllers/submit/landing_page_controller.rb:6]:
```suggestion
```

File: cosmetics-web/spec/features/multiple_responsible_person_spec.rb [hunk 4]
```diff
@@ -1,6 +1,6 @@
 require "rails_helper"
 
-RSpec.describe "Submit user belongs to multiple responsible persons", type: :feature do
+RSpec.describe "Submit user belongs to multiple responsible persons", :with_2fa, :with_stubbed_notify, type: :feature do
```
Comment (scruti) [inline cosmetics-web/spec/features/multiple_responsible_person_spec.rb:3]: The production env runs with 2FA on, tests weren't. The bug was depending on 2FA being enabled so... let's enable it on the specs.

**Comment by nasirkhanbeis** [comment]
Test scenarios:< In-Progress>

1. Verify user  should be able to switch between RPs and go to the home page
2. Verify session time out( remove the 2fa cookie from the browser) and click your cosmetics page. Should prompt the user to enter 2fa code
3. Verify invite user and create a user with the same email account. On 2fa, the user should be prompted to either create new RP or join the existing invite.
4. Verify create account form and invite the user with the same email account. should
5. Verify create an account and before RP creation reset the password- should resend create account link
6. Verify invite user and try reset password with the same account - should resend invite link

**Review by nasirkhanbeis — APPROVED** [review]
Tested this on the review app, looks all good as expected.
Verified this by deleting 2fa cookie in the browser, works fine as expected.
Also tested regression scenarios looking good.

**Review by stdstaff — APPROVED** [review]

**Pull request merged**